### ISSUE TITLE
Repo: snapshot "are clean" checks fail when the regeneration step hits the Nx cache

### ISSUE THREAD

**Issue by Han5991** [typescript-eslint/typescript-eslint#12642]
### AI Assistance

- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Suggestion

The `Verify scope-manager snapshots are clean` / `Verify ast-spec snapshots are clean` checks added in #12623 can fail on branches that have no snapshot problem at all, because the regeneration step goes through the Nx cache.

The job does:

```yaml
- name: Delete all scope-manager fixture snapshots
  run: pnpm exec nx run scope-manager:clean-fixtures
- name: Regenerate scope-manager fixture snapshots
  run: pnpm exec nx test scope-manager -u        # ci.yml#L389-L392
- name: Fail if regenerating snapshots produced a diff
  run: git status --porcelain -- packages/scope-manager/tests/fixtures
```

When step 2 is an Nx cache hit, Nx replays the cached terminal output in 0s without running vitest. `nx.json` declares only `{projectRoot}/coverage` as an output for `test`:

```json
"test": {
  "dependsOn": ["^build"],
  "options": { "config": "vitest.config.mts" },
  "outputs": ["{projectRoot}/coverage"]
}
```

so nothing restores the `.shot` files that step 1 deleted, and step 3 reports every snapshot as deleted.

**Evidence** — the same job, same command, differing only in cache state:

| Run | `scope-manager:test -u` | Result |
| --- | --- | --- |
| [30551756231](https://github.com/typescript-eslint/typescript-eslint/actions/runs/30551756231) (`main`) | `12s` · Cache Miss | pass |
| [30566943775](https://github.com/typescript-eslint/typescript-eslint/actions/runs/30566943775) (`export-ts-globs`) | `0s` · Remote Cache Hit | **fail** |
| [30677180604](https://github.com/typescript-eslint/typescript-eslint/actions/runs/30677180604) (#12218) | `0s` · Remote Cache Hit | **fail** |

The failing logs still print `Snapshots 293 written` — that is replayed cached stdout, not a real run.

Because `-u` is part of the cache key, the `scope-manager:test -u` entry is created by this job itself. It was populated by the first `main` run at 2026-07-30 14:27 (Cache Miss, 12s), and every run since that reaches the job with unchanged `scope-manager` inputs hits it. So this does not self-heal — it is sticky from that point on.

**Proposed fix:** drop the delete-and-regenerate dance and check the file set directly. The check's stated intent is that no `.shot` file is stale/orphaned/missing, and for `scope-manager` that is a 1:1 mapping between each fixture source and `<source>.shot`, so it needs no test run and no cache interaction:

```js
const sources = glob.sync('**/*.{js,ts,jsx,tsx}', {
  cwd: FIXTURES_DIR,
  ignore: ['fixtures.test.ts'],
});
const expected = new Set(sources.map(rel => `${rel}.shot`));
const actual = glob.sync('**/*.shot', { cwd: FIXTURES_DIR });
// fail on: actual.filter(f => !expected.has(f))  (orphans)
//          [...expected].filter(f => !actual.includes(f))  (missing)
```

Snapshot *content* staleness is already caught by the normal unit test job, since vitest fails on mismatch under `CI` and reports obsolete snapshots. What the delete-and-regenerate step adds over that is detection of orphaned files, which is what the comparison above does — deterministically and in ~0s. I ran it against current `main` locally: 293 sources / 293 snapshots / 0 orphans / 0 missing. (The equivalent mapping for `ast-spec` would need checking; I only verified `scope-manager`.)

If a smaller change is preferred, adding `--skip-nx-cache` to both regeneration steps fixes this particular failure mode:

```yaml
run: pnpm exec nx test scope-manager -u --skip-nx-cache
```

I'd suggest *not* declaring the snapshots as `test` outputs instead: they are source-controlled files rather than build artifacts, a cached restore could clobber a contributor's working tree, and `test` vs `test -u` would need different outputs for the same target.

### Additional Info

**Affected runs since the check landed on `main` (`0c94e088e`, 2026-07-30 12:49 UTC):**

| When (UTC) | Branch | Check | Cache state |
| --- | --- | --- | --- |
| 07-30 15:04 | `babel-8` | ast-spec | 7s Cache Miss |
| 07-30 17:39 | `export-ts-globs` | scope-manager | **0s Remote Cache Hit** |
| 07-30 20:07 | `autofix-indentation` | ast-spec | 7s Cache Miss |
| 08-01 00:57 / 01:07 | #12218 | scope-manager | **0s Remote Cache Hit** |

The split is clean: both `scope-manager` failures are cache hits and match the analysis above.

**Possibly separate:** all four `ast-spec` failures (including two on #12623's own branch before merge) happened with a *Cache Miss*, so vitest did run and a diff still appeared. That may be the check correctly catching real stale files — though `babel-8` failed in [30554836748](https://github.com/typescript-eslint/typescript-eslint/actions/runs/30554836748) and passed in [30555397151](https://github.com/typescript-eslint/typescript-eslint/actions/runs/30555397151) under the same Cache Miss condition, which that explanation doesn't cover. I did not investigate this further and it may warrant a separate look.

I noticed this from #12218, where the check fails on a change that only touches `packages/eslint-plugin`. One thing I could not explain is why `scope-manager` is considered affected there at all — the failure mechanism is confirmed, but the trigger condition isn't.

_AI disclosure: the investigation and this write-up were done with AI assistance; every command output and log reference above was verified by me._

**Comment by StyleShit** [comment]
> AI disclosure: the investigation and this write-up were done with AI assistance; every command output and log reference above was verified by me.

It's fine to use AI for investigations, but please write the description [in your own words](https://typescript-eslint.io/contributing/ai-policy/)
